### PR TITLE
Add toast positioning

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Alternative for https://github.com/twbs/bootstrap/pull/28783. Uses our built-in spacing utilities doesn't require JavaScript positioning.

Also added `translate-middle-x` & `translate-middle-y` utilities for more positioning possibilities.

Preview:

* https://deploy-preview-32280--twbs-bootstrap.netlify.app/docs/5.0/components/toasts/#placement
* https://deploy-preview-32280--twbs-bootstrap.netlify.app/docs/5.0/utilities/position/#center-elements

Closes https://github.com/twbs/bootstrap/pull/28783

### CODE REVIEW

File: site/assets/js/application.js [hunk 1]
```diff
@@ -28,13 +28,22 @@
...
+        toastPlacement.dataset.originalClass = toastPlacement.className
+      }
+
+      toastPlacement.className = toastPlacement.dataset.originalClass + ' ' + this.value
+    })
+  }
+
   document.querySelectorAll('.toast')
     .forEach(function (toastNode) {
-      var toast = new bootstrap.Toast(toastNode, {
```
Comment (XhmikosR) [inline site/assets/js/application.js:33]: Can you revert this change please?

**Comment by XhmikosR** [comment]
From a really quick look, I definitely dig it ❤️ 

Needs more eyes from @twbs/css-review

File: site/assets/js/application.js [hunk 2]
```diff
@@ -28,13 +28,22 @@
...
     })
 
+  var toastPlacement = document.getElementById('toastPlacement')
+  if (toastPlacement) {
+    document.getElementById('selectToastPlacement').addEventListener('change', function () {
+      if (!toastPlacement.dataset.originalClass) {
+        toastPlacement.dataset.originalClass = toastPlacement.className
+      }
+
+      toastPlacement.className = toastPlacement.dataset.originalClass + ' ' + this.value
```
Comment (XhmikosR) [inline site/assets/js/application.js:38]: I assume you want to overwrite any previous classes here? If not, we could just use `classList.add()`?

Comment (MartijnCuppens) [reply site/assets/js/application.js:38]: Yep, the class needs to be overridden since position utilities can be added.

File: site/assets/js/application.js (hunk 1 above)
Comment (MartijnCuppens) [reply site/assets/js/application.js:33]: Done

**Comment by XhmikosR** [comment]
BTW the center widgets look weird on mobile

image

**Comment by MartijnCuppens** [comment]
> BTW the center widgets look weird on mobile

Fixed that

**Review by mdo — APPROVED** [review]
Let's do it!

**Comment by XhmikosR** [comment]
@mdo we have #28783 included in beta1. I'm not sure if these are 1 to 1 equivalent, but if so, I guess this one should be merged and not #28783.

**Comment by mdo** [comment]
Widths need to be ironed out I think... by default, the toasts are shrunk here. Is that behavior we want overall? Would we want to change to this behavior at all times and let folks pick their own `max-width`s?

Screen Shot 2020-12-02 at 10 24 15 AM

It's not until I add wrapping content in one that they hit their `max-width` that's set on the `.toast`.

Screen Shot 2020-12-02 at 10 22 33 AM

File: scss/_toasts.scss [hunk 3]
```diff
@@ -2,31 +2,30 @@
...
 
   &.hide {
     display: none;
   }
 }
 
+.toast-container {
+  display: grid;
+  width: max-content;
+  max-width: 100%;
```
Comment (mdo) [inline scss/_toasts.scss:24]: Drop these two lines for now? We can add them later if we really need them, but I think we can let our toasts stay at their `max-width`.

```suggestion
```

**Comment by mdo** [comment]
Also, could this arguably wait until another release? Not sure we need to lock in something for the beta release.

**Comment by MartijnCuppens** [comment]
Good point, hadn't noticed that. We'll need the `width: max-content;` to fix https://github.com/twbs/bootstrap/pull/32280#issuecomment-735437987

Just fixed the shrinking thing by using margins between the toasts instead of the using grid.

**Review by mdo — APPROVED** [review]
Thank you!

**Comment by XhmikosR** [comment]
Branch rebased and replaced the renamed classes.

@ffoodd RTL-wise does this LGTY?

**Review by ffoodd — APPROVED** [review]
LGTM!

Since position relies on our utilities, that's already handled and processed by RTLCSS :+1: 

Regarding cheatsheet, I don't think so but that soemthing to question: I didn't show utilities in the cheatsheet for now. I'll add it to our remaining to-do.
